src/main/java/thePackmaster/cards/magnetizepack/Attraction.java

```diff
@@ -0,0 +1,44 @@
+package thePackmaster.cards.magnetizepack;
+
+import basemod.BaseMod;
+import basemod.helpers.CardModifierManager;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInHandAction;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import thePackmaster.actions.magnetizepack.MagnetizeAction;
+import thePackmaster.cardmodifiers.magnetizepack.MagnetizedModifier;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+public class Attraction extends AbstractMagnetizeCard {
+    public final static String ID = makeID(Attraction.class.getSimpleName());
+
+    public Attraction() {
+        super(ID, 1, CardType.SKILL, CardRarity.UNCOMMON, CardTarget.NONE);
+        baseMagicNumber = magicNumber = 3;
+        cardsToPreview = new Fuzz();
+    }
+
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        for (int i = 0; i < magicNumber; i++) {
+            addToBot(new DrawCardAction(1));
+            if (p.hand.size() + i <= BaseMod.MAX_HAND_SIZE)
```

Comment:
This approach could be mislead by things drawing inbetween these actions which could be done via beacon iirc or other modded relics

Might be a better idea to make into an anonymous action that checks if the hand isn't full. Basically jsut combine this check with the anonymous action you use to apply magentize